packages/semcom-node/lib/store/services/component-in-memory-store.service.ts

```diff
@@ -7,16 +8,27 @@ export class ComponentInMemoryStore extends ComponentStore {
   }
 
   async query(filter: Partial<ComponentMetadata>): Promise<ComponentMetadata[]> {
-    return this.components.filter((component) =>
+    let filtered = this.components.filter((component) =>
       Object.keys(filter).every((key) => {
         const componentValue = component[key];
         const filterValue = filter[key];
         return (componentValue === filterValue) || (
           Array.isArray(componentValue) &&
           Array.isArray(filterValue) &&
           filterValue.every((value) => componentValue.includes(value))
+          || (key === 'version')
         );
       }));
+
+    // seperated version logic for readability
+
+    if(filtered && filter.version) {
+      const versions = filtered.map((component) => component.version);
+      const maxVersion = semver.maxSatisfying(versions, filter.version);
+      filtered = [ filtered.find((component) =>  component.version === maxVersion) ];
+    }
```

Comment:
Rewrite this so it does not return undefined (e.g. using filter instead of find).